metadata.pl

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env perl
+#--------------------------------------------------------------
+# metadata.pl: i/o for json, yaml, and run.properties files
+#--------------------------------------------------------------
+# Copyright(C) 2021 Jason Fleming
+# 
+# This file is part of the ADCIRC Surge Guidance System (ASGS).
+# 
+# The ASGS is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# ASGS is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with the ASGS.  If not, see <http://www.gnu.org/licenses/>.
+#--------------------------------------------------------------
+# This script accepts one or more keys whose values will be read 
+# from the specified metadata file; these will be returned as 
+# a set of key-value pairs.
+#
+# Alternatively, it will accept a string of key-value pairs to
+# be added to the metadata file.   
+#
+# The script determines the format of the metadata file 
+# automatically using the file extension (.json, .yaml, or 
+# .properties).
+#--------------------------------------------------------------
+$^W++;
```

Comment:
Just for future reference, this can be accomplished with `use warnings;` - just to make it look similar to `use strict;`.